### PR TITLE
ACS-4766 Migrating to GH Actions 5.0.N

### PR DESCRIPTION
**Pull request by mikolajbrzezinski**

### CODE REVIEW

**Comment by mikolajbrzezinski** [comment]
Translated as-is on Travis(I think), but I doubt it was in a good state (https://app.travis-ci.com/github/Alfresco/alfresco-java-sdk/builds/252769285). Could be looked into more but don't know if its in the scope of this ticket

**Review by damianujma — APPROVED** [review]

**Comment by dsibilio** [comment]
> 

What do you mean by "it was not in a good state". What's the effects we're seeing and are concerned about?

**Comment by mikolajbrzezinski** [comment]
I believe that the output for the "build application" job on Travis is very lackluster and, at least for me, it seems to not be correctly set up to test the repo. Compare it to the output from later iterations of the repo @dsibilio

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      - build_and_verify
+    env:
+      MAVEN_CLI_OPTS: >
+        source:jar -B -q -e -fae -V -DinstallAtEnd=true -U -Ddoclint=none
+        -DaltReleaseDeploymentRepository=alfresco-public::default::https://artifacts.alfresco.com/nexus/content/repositories/releases
+        -DaltSnapshotDeploymentRepository=alfresco-public-snapshots::default::https://artifacts.alfresco.com/nexus/content/repositories/snapshots
+    steps:
+      - uses: actions/checkout@v3
+      - uses: Alfresco/alfresco-build-tools/.github/actions/get-build-info@v1.35.0
+      - uses: Alfresco/alfresco-build-tools/.github/actions/setup-java-build@v1.35.0
```
Comment (dsibilio) [inline .github/workflows/ci.yml]: Let's update this to `1.36.0` (same in other places within the workflow), otherwise looks good.

**Review by dsibilio — APPROVED** [review]

**Comment by dsibilio** [comment]
> I believe that the output for the "build application" job on Travis is very lackluster and, at least for me, it seems to not be correctly set up to test the repo. Compare it to the output from later iterations of the repo @dsibilio

Commands look very similar if not identical to me, it might just be a difference in the log level configuration for some of the tests. I think it's fine to leave it as-is.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mikolajbrzezinski) [reply .github/workflows/ci.yml]: Done

**Review by aonych — APPROVED** [review]

**Review by dsibilio — CHANGES_REQUESTED** [review]
Please adapt the `ciManagement` section as intended. You can check https://github.com/Alfresco/alfresco-elasticsearch-connector/blob/master/pom.xml#L41-L44 as an example.

**Pull request merged**